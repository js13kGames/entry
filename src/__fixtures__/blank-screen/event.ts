import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 1,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/blank-screen",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "360b838";
