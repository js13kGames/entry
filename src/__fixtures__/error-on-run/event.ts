import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 6,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/error-on-run",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;
