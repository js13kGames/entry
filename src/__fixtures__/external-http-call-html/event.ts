import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 5,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/external-http-call-html",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "c9a43c6";
