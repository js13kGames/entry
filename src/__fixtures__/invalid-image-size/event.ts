import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 12,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/invalid-image-size",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "ce740ea";
