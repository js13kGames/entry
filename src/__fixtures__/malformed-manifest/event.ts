import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 10,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/malformed-manifest",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;
