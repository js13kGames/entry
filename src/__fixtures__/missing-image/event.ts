import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 11,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/missing-image",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "400d963";
