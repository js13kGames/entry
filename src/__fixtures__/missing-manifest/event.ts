import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 9,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/missing-manifest",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "de7f563";
