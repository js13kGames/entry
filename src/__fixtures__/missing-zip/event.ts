import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 3,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/missing-zip",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "e60be98";
