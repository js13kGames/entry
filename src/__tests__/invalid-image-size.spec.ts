import * as event from "../__fixtures__/invalid-image-size/event";
import { bootstrap } from "./_bootstrap";

describe("integration  invalid-image-size", () => {
  bootstrap(event);
});
