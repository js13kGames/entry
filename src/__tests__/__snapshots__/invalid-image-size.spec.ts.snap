// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration  invalid-image-size should analyze 1`] = `
Array [
  Object {
    "conclusion": "success",
    "files": Array [
      "asset/image/1659821.jpeg",
      "asset/script.js",
      "asset/text.txt",
      "image_large.jpg",
      "image_thumbnail.jpg",
      "index.html",
      "manifest.json",
    ],
    "name": "manifest-found",
  },
  Object {
    "conclusion": "success",
    "manifest": Object {
      "description": "This entry should yield an error because one image does not meet the ratio required",
      "name": "invalid image size",
    },
    "name": "manifest-read",
  },
  Object {
    "conclusion": "success",
    "gameName": "invalid image size",
    "name": "name-found",
  },
  Object {
    "conclusion": "success",
    "gameDescription": "This entry should yield an error because one image does not meet the ratio required",
    "name": "description-found",
  },
  Object {
    "conclusion": "failure",
    "images": Object {
      "image_large": Object {
        "error": "size",
        "origin": Object {
          "height": 300,
          "width": 200,
        },
        "target": Object {
          "height": 200,
          "width": 300,
        },
      },
      "image_thumbnail": "https://res.cloudinary.com/dy3iszldp/image/upload/c_fit,h_100,w_100/61191dccc4c29b4d785cae9b2f2eae2d.jpg",
    },
    "name": "images-found",
  },
  Object {
    "conclusion": "success",
    "name": "release-found",
    "tagName": "invalid-image-size-v1",
  },
  Object {
    "assetFiles": Array [
      "bundle.zip",
    ],
    "conclusion": "success",
    "name": "bundle-found",
  },
  Object {
    "conclusion": "success",
    "name": "bundle-unzipped",
  },
  Object {
    "bundleSize": 3859,
    "conclusion": "success",
    "name": "bundle-size",
    "sizeLimit": 13312,
  },
  Object {
    "bundleFiles": Array [
      "asset/text.txt",
      "asset/script.js",
      "asset/image/1659821.jpeg",
      "index.html",
    ],
    "conclusion": "success",
    "deployUrl": "http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/cec73ecd46375ab7c64a85c8a4d2c172/index.html",
    "name": "index-found",
  },
  Object {
    "conclusion": "success",
    "errors": Array [],
    "name": "run-without-error",
  },
  Object {
    "conclusion": "success",
    "externalUrls": Array [],
    "name": "run-without-external-http",
    "urls": Array [
      "http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/cec73ecd46375ab7c64a85c8a4d2c172/asset/image/1659821.jpeg",
      "http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/cec73ecd46375ab7c64a85c8a4d2c172/asset/script.js",
      "http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/cec73ecd46375ab7c64a85c8a4d2c172/asset/text.txt",
      "http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/cec73ecd46375ab7c64a85c8a4d2c172/index.html",
      "http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/favicon.ico",
    ],
  },
  Object {
    "conclusion": "success",
    "name": "run-without-blank-screen",
    "screenShotUrl": "http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/cec73ecd46375ab7c64a85c8a4d2c172/game_preview_screenshot.png",
  },
]
`;

exports[`integration  invalid-image-size should generate report 1`] = `
"Hello üëã
I am a bot. My job is to review your submission.

I reviewed your latest release, [invalid-image-size-v1](https://github.com/Platane/entry/releases/tag/invalid-image-size-v1). Its looking all good üëç

The image \`image_large\` does not meet the requirement, it should have a ratio of 300x200 (instead of 200x300)

<blockquote> <br> <img height=\\"80\\" width=\\"80\\" title=\\"thumbnail\\" src=\\"https://res.cloudinary.com/dy3iszldp/image/upload/c_fit,h_100,w_100/61191dccc4c29b4d785cae9b2f2eae2d.jpg\\" /> <div>  <a href=\\"http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/cec73ecd46375ab7c64a85c8a4d2c172/index.html\\">   <strong>invalid image size</strong>  </a>  <br>  This entry should yield an error because one image does not meet the ratio required </div> <br></blockquote>"
`;
